db/memtable.h

```diff
@@ -387,6 +389,12 @@ class MemTable {
 
   Env* env_;
 
+  // Extract sequential insert prefixes.
+  const SliceTransform* insert_prefix_extractor_;
+
+  // Insert hints for each prefix.
+  std::unordered_map<Slice, MemTableInsertHint, SliceHasher> insert_hints_;
```

Comment:
Would be nice to account for the memory used by this map in WriteBufferManager and MemTable::ApproximateMemoryUsage(). Maybe it's possible to use the same ConcurrentArena allocator for this map as for the memtable contents (with a no-op deallocate(), with an assumption that unordered_map doesn't deallocate much); that would automatically count the memory usage in both places, and may also improve performance.